linkerd/app/inbound/src/direct.rs

```diff
@@ -449,6 +449,12 @@ impl Param<profiles::LookupAddr> for GatewayTransportHeader {
     }
 }
 
+impl Param<linkerd_proxy_client_policy::TargetAddr> for GatewayTransportHeader {
+    fn param(&self) -> linkerd_proxy_client_policy::TargetAddr {
+        linkerd_proxy_client_policy::TargetAddr(self.target.clone().into())
+    }
+}
+
```

Comment:
This should be in the gateway and not in inbound. The inbound proxy doesn't care how client policies are looked up from these headers.